### PR TITLE
Gjt contam centroid review

### PR DESCRIPTION
**Pull request by talensgj**
Mostly done, the new algorithm is now in dms/soss_centroids.py. I'm looking for feedback on the remaining TODO items (and related the slightly out-of-control logic blocks in those functions.) Also on the 2 expected type errors in PyCharm. 

Aside from that I think for clarity it might be good to rename the previous 2 algorithms (mostly for reasons of line-length economy):
get_uncontaminated_centroids() -> get_centroids_com()
get_uncontaminated_centroids_edgetrigger() -> get_centroids_edgetrigger()

### CODE REVIEW

**Comment by talensgj** [comment]
Made small modifications following feedback from @loicalbert

File: SOSS/dms/soss_centroids.py [hunk 1]
```diff
@@ -321,7 +394,7 @@ def get_uncontam_centroids_edgetrig(image, header=None, mask=None, poly_order=11
     if mask is None:
         mask = np.zeros_like(image, dtype='bool')
 
-    # Call the script that determines the dimensions of the stack.
+    # Call the script that determines the dimensions of the image.
     dimx, dimy, xos, yos, xnative, ynative, padding, refpix_mask = \
         get_image_dim(image, header=header, verbose=verbose)
```
Comment (njcuk9999) [inline SOSS/dms/soss_centroids.py:399]: I don't like the `\` operator 
you could do this:
```python
gout = get_image_dim(image, header=header, verbose=verbose)
dimx, dimy, xos, yos, xnative, ynative, padding, refpix_mask = gout
```

or if line 2 is still too long:
```python
gout = get_image_dim(image, header=header, verbose=verbose)
dimx, dimy, xos, yos, xnative = gout[:5]
ynative, padding, refpix_mask = gout[5:]
```

File: SOSS/dms/soss_centroids.py [hunk 2]
```diff
@@ -362,6 +435,860 @@ def get_uncontam_centroids_edgetrig(image, header=None, mask=None, poly_order=11
...
+    :param halfwidth_o3: the size of the window to mask around the order 3 trace.
+    :param xlim: the boundary for masking pixels blue-ward (to the right).
+
+    :type ytrace_o1: array[float]
+    :type ytrace_o3: array[float]
+    :type subarray: str
+    :type halfwidth_o1: float
+    :type halfwidth_o3: float
+    :type xlim: float
+
```
Comment (njcuk9999) [inline SOSS/dms/soss_centroids.py:676]: you can actually add a "raises" flag in the doc stringv

```python
:raises ValueError: if subarray not either 'FULL', 'SUBSTRIP96', 'SUBSTRIP256'
```

only really useful for others using this function - that they may need to catch this exception - in this case probably isn't that useful - but extra info is always good

**Review by njcuk9999 — APPROVED** [review]
Very nice and clean - can't find much at all I'd change
added two general suggestions on about avoiding the use of `/`  - you can always do something else and the other about adding `:raises ValueError: message` to doc strings to tell users / other devs that this function can raise an error (can be very helpful telling people they may need to catch a possible specific exception) - again both just suggestions!

**Review by loicalbert — APPROVED** [review]
All good

**Pull request merged**